OpenRA.Game/Traits/ActivityUtils.cs

```diff
@@ -44,7 +44,7 @@ public static Activity RunActivity(Actor self, Activity act)
 				else
 					start = current;
 
-				if (prev == act)
+				if (act == prev || act == prev.Parent)
```

Comment:
The reason I went that way was to avoid touching (and auditing!) all the existing activities.

You don't normally need to know what `NextActivity` will return anyway.

Also, if you insist on undoing this "uncertainty" in the future, I'm pretty sure that the `CompositeActivity` thing will stop to work, since it relies on exactly that ambiguity to work.